### PR TITLE
CMake gtest/benchmark deps refactor + CI deps change

### PR DESCRIPTION
**Pull request by kaihsin**
1. remove google-benchmark
2. change gtest to access from deps instead of build from scrach 
3. change support python version to 3.9/.10/.11 for both linux and mac (#281)
4. change all ci running python to 3.11

### CODE REVIEW

**Comment by kaihsin** [comment]
@hunghaoti @j9263178  make sure you all on the same page that we remove benchmark deps, so if you are working on test cases, don't use benchmark framework anymore

File: .github/workflows/coverity-scan.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
     - name: Install dependencies
       shell: bash -l {0}
       run: |
-        mamba install _openmp_mutex=*=*_llvm cmake make boost git compilers numpy mkl mkl-include mkl-service pybind11 libblas=*=*mkl
+        mamba install _openmp_mutex=*=*_llvm cmake make boost git compilers numpy mkl mkl-include mkl-service pybind11 libblas=*=*mkl gtest
```
Comment (hunghaoti) [inline .github/workflows/coverity-scan.yml:36]: We may need to update this command to user guide for 'advanced install' part. (add gtest)

**Review by hunghaoti — APPROVED** [review]
We may need to update the following command in user guide in the part 'advanced install' (add gtest)
```bash
$conda install cmake make boost libboost git compilers numpy mkl mkl-include mkl-service pybind11 libblas=*=*mkl beartype gtest
```

**Comment by hunghaoti** [comment]
> We may need to update the following command in user guide in the part 'advanced install' (add gtest)
> 
> ```shell
> $conda install cmake make boost libboost git compilers numpy mkl mkl-include mkl-service pybind11 libblas=*=*mkl beartype gtest
> ```

Added in dev-master branch.

**Pull request merged**